File: scripts/codereview/testdata/py/before.py
"""Example module for testing AST extraction."""

import os
from typing import Optional, List
from dataclasses import dataclass


@dataclass
class User:
    id: int
    name: str


class UserService:
    """Service for managing users."""

    def __init__(self, db_url: str):
        self.db_url = db_url

    def get_user(self, user_id: int) -> Optional[User]:
        """Get a user by ID."""
        return None

    def list_users(self) -> List[User]:
        """List all users."""
        return []


def greet(name: str) -> str:
    """Return a greeting message."""
    return f"Hello, {name}!"


async def fetch_data(url: str) -> dict:
    """Fetch data from a URL."""
    return {}


def format_name(name: str) -> str:
    """Format a name."""
    return name.strip().title()

File: scripts/codereview/testdata/go/before.go
package example

import (
	"fmt"
	"strings"
)

// User represents a user in the system
type User struct {
	ID   int
	Name string
}

// Greeter interface for greeting
type Greeter interface {
	Greet() string
}

// Hello returns a greeting message
func Hello(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// (u *User) GetName returns the user's name
func (u *User) GetName() string {
	return u.Name
}

// FormatName formats a name with optional prefix
func FormatName(name string) string {
	return strings.Title(name)
}
